Engine/Engine/Core/AssetManager.cpp

```diff
@@ -27,8 +27,16 @@ AssetManager::AssetManager()
  */
 boost::shared_ptr<MeshData> AssetManager::GetMesh(std::string id) const
 {
-	if(this->meshes.find(this->meshShortNames.find(id)->second) != this->meshes.end())
-		return this->meshes.find(this->meshShortNames.find(id)->second)->second;
+	if(this->meshShortNames.find(id) != this->meshShortNames.end())
+	{
+		if(this->meshes.find(this->meshShortNames.find(id)->second) != this->meshes.end())
+			return this->meshes.find(this->meshShortNames.find(id)->second)->second;
+	}
+
+	std::string msg = "Asset with ID '";
```

Comment:
Why do we have so much `std::string` in the codebase when we're using boost? Wouldn't it be more appropriate to use `boost::container::string`?